lib/Service/Provisioning/Manager.php

```diff
@@ -274,6 +274,21 @@ private function updateAccount(IUser $user, MailAccount $account, Provisioning $
 		return $account;
 	}
 
+	private function isUpToDate(IUser $user, MailAccount $account, Provisioning $provisioning): bool {
```

Comment:
instead of a custom check, how about we use `\OCP\AppFramework\Db\Entity::getUpdatedFields` after the account was updated with the new/current values. If all properties have the same value as before, the mapper will not have to update anything.

Reply:
I'm actually thinking that I had that in mind. If the account has no changes, the mapper does a noop. Could you debug why this isn't the case anymore? From the blackfire.io session I remember that the UPDATE query assigned a new provisioning_id, right?

Reply:
*Your* bug https://github.com/nextcloud/mail/blame/27fe2975aac226aad344d7a54b0858d136565d91/lib/Db/MailAccount.php#L199

Reply:
Ah. The good old snake_case / camelCase mixup ;)